fvp-tpm.mk

```diff
@@ -0,0 +1,241 @@
+################################################################################
```

Comment:
This file has quite a bit in common with fvp.mk, would it be possible to merge them and only set some flag to enable the needed tpm stuff? It would be easier to maitain.

Reply:
Definitely, I can take a look at it

Reply:
Hi @jenswi-linaro,

Before I submit my changes, I created an fvp-common.mk file, which is mostly a copy of fvp.mk. I modified the latter so it  now includes the former and just defines a couple of environment variables and the main targets.

I did the same for fvp-tpm.mk: it includes the common file and defines its own environment variables and targets so it is simpler than it was before. There are no duplicates as they all are now in the common file.

Would that be acceptable?

Cheers,
Javier

Reply:
It's an improvement, but wouldn't it be even better to only have it as a config option to the normal FVP setup? That way you can share the manifest.

Reply:
I actually thought about that but I wasn't sure if modifying an existing platform would be accepted. But if it is, I can add the ftpm and tss on fvp.mk for all cases (they barely introduce any overhead) and add a build option to enable Measured Boot on TF-A if required (which is the only option that could cause any problem).

Reply:
I second Jens here, I think it's preferred having it in the already existing fvp.mk file.

Reply:
I second Jens here, I think it's preferred having it in the already existing fvp.mk file.

Reply:
Hi @jenswi-linaro 

I pushed a new version of the patch, adding my functionality to fvp.mk rather than creating a new platform. I introduced the USE_MEASURED_BOOT build variable to enable measured boot (by default I rather keep TF-A with the default options than build it with Measured Boot as that feature is memory consuming).

Thanks,
Javier